lib/datadog/appsec/processor.rb

```diff
@@ -38,9 +36,30 @@ def run(input, timeout = WAF::LibDDWAF::DDWAF_RUN_TIMEOUT)
           @run_mutex.unlock
         end
 
+        def extract_schema
+          return unless extract_schema?
+
+          input = {
+            'waf.context.processor' => {
+              'extract-schema' => true
+            }
+          }
+
+          _code, res = @context.run(input, WAF::LibDDWAF::DDWAF_RUN_TIMEOUT)
```

Comment:
This `libddwaf` API looks _very_ strange, which might be the root cause of my confusion.